packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/stage-parser.ts

```diff
@@ -36,13 +36,14 @@ export function assertStageNode(node: t.Node): asserts node is StageLike {
   if (isStageLike(node)) {
     return;
   }
-  throw new SyntaxError(
-    node.type === 'ObjectExpression'
-      ? (node.properties[0] as t.ObjectProperty | undefined)?.key == null
-        ? 'A pipeline stage specification object must contain exactly one field.'
-        : 'Stage value can not be empty'
-      : 'Each element of the pipeline array must be an object'
-  );
+  const { column, line } = node.loc?.start ?? { line: 1, column: 1 };
```

Comment:
Can't say I have a strong preference, but I wonder whether we even want a fallback if there is not loc info available on the node here 🤔 Is 1:1 a meaningful fallback also? Seems like babel when fails to parse for example always defaults to the end of file

Reply:
removed the fallback option. 
now that we are showing syntax errors as warnings on screen, this should not be an issue.